gdx/test/com/badlogic/gdx/math/QuaternionTest.java

```diff
@@ -0,0 +1,23 @@
+
+package com.badlogic.gdx.math;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class QuaternionTest {
+	private static final float epsilon = Float.MIN_NORMAL;
```

Comment:
Is `MathUtils.FLOAT_ROUNDING_ERROR` too low here?

Reply:
We could try it, that's a good idea. I think the Math.atan2() version will pass FLOAT_ROUNDING_ERROR but not MIN_NORMAL (which is incredibly low, 1.17549435E-38f). We know the all-double version can pass Float.MIN_NORMAL, though I don't think it can pass Double.MIN_NORMAL simply because floats can't go that low without being subnormal (maybe even then) or 0.

Reply:
Nope, neither passes FLOAT_ROUNDING_ERROR. If I change `MathUtils.atan2()` to `(float)Math.atan2()`, I get

```
Expected :-5.0
Actual   :-5.0000014
```

Which is of course very close to where it needs to be, but not quite. If I didn't change MathUtils' atan2, I would have:

```
Expected :-5.0
Actual   :-5.0020123
```

And if I use my same-speed approximation (sheet 11), I get:
```
Expected :-5.0
Actual   :-5.0000577
```

The error on my approximation is 40x worse than (float)Math.atan2(), though Math should be very nearly 0 error, and the method should usually be about 4x faster than (float)Math.atan2(). The error on the current MathUtils.atan2() is about 1400x worse than (float)Math.atan2().

...And none pass with MathUtils.FLOAT_ROUNDING_ERROR as the threshold, so maybe we need to do the double-precision quaternions anyway.

Reply:
I've updated the epsilon to `0.0001f` as suggested.